tests/foreman/ui/test_contentview.py

```diff
@@ -4112,10 +4521,84 @@ def test_positive_delete_cv_promoted_to_multi_env(self):
+        all_env_names_set.add(ENVIRONMENT)
+        puppet_repo_url = FAKE_0_PUPPET_REPO
+        puppet_module_name = FAKE_0_PUPPET_MODULE
+        repos = [
+            dict(
+                repo_name=gen_string('alpha'),
+                repo_url=FAKE_0_YUM_REPO,
+                repo_type=REPO_TYPE['yum'],
+            ),
+            dict(
+                repo_name=gen_string('alpha'),
+                repo_url=puppet_repo_url,
+                repo_type=REPO_TYPE['puppet'],
+            ),
+        ]
+        # create a new organization
+        org = entities.Organization().create()
+        with Session(self.browser) as session:
+            # create the DEV, QE, STAGE, PROD lifecycle environments
+            env_prior = None
+            for env_name in env_names:
+                make_lifecycle_environment(
+                    session, org=org.name, name=env_name, prior=env_prior)
+                env_prior = env_name
+            # create the repositories
+            for repo in repos:
+                self.setup_to_create_cv(
+                    org_id=org.id,
+                    **repo
+                )
+            # create a content view
+            make_contentview(
+                session, org=org.name, name=cv_name)
+            self.assertIsNotNone(self.content_views.search(cv_name))
+            # add the repos to content view
+            for repo in repos:
+                repo_name = repo['repo_name']
+                repo_type = repo['repo_type']
+                if repo_type == 'puppet':
+                    self.content_views.add_puppet_module(
+                        cv_name,
+                        puppet_module_name,
+                        filter_term='Latest',
+                    )
+                    self.assertIsNotNone(
+                        self.content_views.fetch_puppet_module(
+                            cv_name, puppet_module_name)
+                    )
+                else:
+                    self.content_views.add_remove_repos(
+                        cv_name, [repo_name], repo_type=repo_type)
+                    self.assertIsNotNone(self.content_views.wait_until_element(
+                        common_locators['alert.success_sub_form']))
+            # publish the content view
+            version = self.content_views.publish(cv_name)
+            self.assertIsNotNone(
+                self.content_views.wait_until_element(
+                    common_locators['alert.success_sub_form'])
+            )
+            # promote the content view to all lifecycle environments
+            for env_name in env_names:
+                self.content_views.promote(cv_name, version, env_name)
+            self.assertEqual(
+                set(self._get_cv_version_environments(version)),
+                all_env_names_set
+            )
+            # remove all the environments from version
+            self.content_views.remove_version_from_environments(
```

Comment:
yes you are right, here the cv cannot be removed when promoted to lcves, need a negative test of this